inc/Engine/Saas/Admin/AdminBar.php

```diff
@@ -0,0 +1,159 @@
+<?php
+declare(strict_types=1);
+
+namespace WP_Rocket\Engine\Saas\Admin;
+
+use WP_Admin_Bar;
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Engine\Common\Context\ContextInterface;
+
+class AdminBar {
+	/**
+	 * Options data instance.
+	 *
+	 * @var Options_data
+	 */
+	private $options;
+
+	/**
+	 * ATF context.
+	 *
+	 * @var ContextInterface
+	 */
+	private $atf_context;
+
+	/**
+	 * RUCSS optimize url context.
+	 *
+	 * @var ContextInterface
+	 */
+	private $rucss_url_context;
+
+	/**
+	 * Constructor
+	 *
+	 * @param Options_Data     $options Options data instance.
+	 * @param ContextInterface $atf_context ATF context.
+	 * @param ContextInterface $rucss_url_context RUCSS optimize url context.
+	 */
+	public function __construct( Options_Data $options, ContextInterface $atf_context, ContextInterface $rucss_url_context ) {
+		$this->options           = $options;
+		$this->atf_context       = $atf_context;
```

Comment:
I think we can change this class a bit to be like the [JobProcessor](https://github.com/wp-media/wp-rocket/blob/task/lcp-refactor-rucss-api-cron-queue/inc/Engine/Common/JobManager/JobProcessor.php) to use factories so it's easy to for future SaaS features to extend it later on that way we will not need to pass individual feature dependencies to it. We can leverage on the factories already created for ATF and RUCSS here.

Reply:
I agree, we could update this after merging the PR with the rest, it will probably be simpler than duplicate what you did on the other PR. Do you agree?